lib/array_equals.rb

```diff
@@ -1,5 +1,24 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  elsif array1.length != array2.length
+    return false
+  elsif array1.length == array2.length
```

Comment:
No need to check the lengths of the array here. If your fn gets to this line you already know the arrays have the same length.